tokio/src/runtime/task/mod.rs

```diff
@@ -204,6 +204,7 @@ pub(crate) struct LocalNotified<S: 'static> {
 /// This type holds two ref-counts.
 pub(crate) struct UnownedTask<S: 'static> {
     raw: RawTask,
+    is_mandatory: bool,
     _p: PhantomData<S>,
 }
```

Comment:
I suppose it makes sense to put it here, but maybe we should either rename the struct to something like `BlockingTask` or put the boolean in a wrapper struct?

Reply:
I've moved things around a bit. I took your advice of defining a wrapper struct that contains a `UnownedTask` and a `is_mandatory` field. The wrapper struct is defined in `pool.rs`, next to where it's used. Unit tests that use `unowned` can keep using it without caring about the `is_mandatory` field.

(I've also changed `is_mandatory: bool` to a `mandatory: Mandatory` where `Mandatory` is a new enum. I did this to make the call-site of code using this parameter easier to read)